File: LinkedList/java/Problem1290_ConvertBinaryNumberInLinkedListToInteger.java
/**
 * Problem 1290: Convert Binary Number in a Linked List to Integer
 * Difficulty: Easy
 * 
 * Convert binary number represented as linked list to integer.
 * Example: [1,0,1] → 5
 * 
 * Time: O(n), Space: O(1)
 */
class ListNode {
    int val;
    ListNode next;
    ListNode(int val) { this.val = val; }
}

public class Problem1290_ConvertBinaryNumberInLinkedListToInteger {
    public int getDecimalValue(ListNode head) {
        int result = 0;
        
        while (head != null) {
            result = result * 2 + head.val;
            head = head.next;
        }
        
        return result;
    }
}

File: LinkedList/python/problem_1290_convert_binary_number_in_linked_list_to_integer.py
"""Problem 1290: Convert Binary Number in Linked List to Integer - Easy
Convert binary linked list to integer
Example: [1,0,1] → 5
Time: O(n), Space: O(1)
"""
class ListNode:
    def __init__(self, val=0, next=None):
        self.val, self.next = val, next

class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        result = 0
        
        while head:
            result = result * 2 + head.val
            head = head.next
        
        return result
